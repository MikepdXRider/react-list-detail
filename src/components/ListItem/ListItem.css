@import url('https://fonts.googleapis.com/css2?family=Pangolin&display=swap');

.container {
    width: 280px;
    max-height: 300px;
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 3fr 1fr;
    border-radius: 10px;
    /* animation: wipe-enter 1s; */

    color: white;
    background: rgba( 255, 255, 255, 0.4 );
    box-shadow: 0 8px 32px 0 rgba( 31, 38, 135, 0.37 );
    backdrop-filter: blur( 10px );
    -webkit-backdrop-filter: blur( 10px );
    border-radius: 10px;
    border: 1px solid rgba( 255, 255, 255, 0.18 );
}

.container img{
    grid-row: 1;
    width: 278px;
    height: 225px;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
}

.cardTitle{
    font-family: 'Pangolin', cursive;
}

.container div{
    grid-row: 2;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    margin-left: 5px;
}

.container span{
    color: purple;
}

.cardAff{
    font-family: 'Coming Soon', cursive;
}

@keyframes wipe-enter {
	0% {
		transform: scale(0, .025);
	}
	50% {
		transform: scale(1, .025);
	}
}

.container:hover{
    cursor: pointer;
    background: rgba( 255, 255, 255, 0.2 );
    box-shadow: 0 8px 32px 0 rgba( 31, 38, 135, 0.37 );
    backdrop-filter: blur( 10px );
    -webkit-backdrop-filter: blur( 10px );
    border-radius: 10px;
    border: 1px solid rgba( 255, 255, 255, 0.18 );
}