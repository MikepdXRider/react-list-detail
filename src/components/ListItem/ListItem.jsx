// Presentational component
// Recieves state, props and functions from container component
// Displays summary of a single list item.

import React from 'react'
import consistentString from '../../utils/consistentString.js'
import styles from './ListItem.css'

export default function ListItem({charName, charPhoto, charAff, charId, onClick}) {
    return (
        // will be grid
            // has background
        <article className={styles.container} onClick={() => onClick(charId)}>
            <img src={`${charPhoto}`} alt={`${charName}`} />
            <div>
                <p className={styles.cardTitle}>{charName}</p>
                    <p 
                    className={styles.cardAff}
                    >
                        {consistentString(charAff)}
                        <span>see more</span>
                    </p>
            </div>
        </article>
    )
}
