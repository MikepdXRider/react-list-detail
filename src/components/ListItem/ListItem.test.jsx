import { render } from "@testing-library/react";
import ListItem from "./ListItem.jsx";

const testObj = {
    charName: 'test-name',
    charPhoto: 'test-photo',
    charAff: 'test-aff',
    charId: 'test-id',
    onClick: jest.fn()
}

it('renders ListItem component', () => {
    const {container} = render(
        <ListItem 
        {...testObj}
        />
    )

    expect(container).toMatchSnapshot();
})