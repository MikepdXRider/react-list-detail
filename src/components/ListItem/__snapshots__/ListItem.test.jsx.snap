// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ListItem component 1`] = `
<div>
  <article
    class="container"
  >
    <img
      alt="test-name"
      src="test-photo"
    />
    <div>
      <p
        class="cardTitle"
      >
        test-name
      </p>
      <p
        class="cardAff"
      >
        test-aff... 
        <span>
          see more
        </span>
      </p>
    </div>
  </article>
</div>
`;
